src/Swarm/Language/Syntax.hs

```diff
@@ -965,3 +966,17 @@ mapFree1 x f = fvT %~ (\t -> if t == TVar x then f t else t)
 
 lowShow :: Show a => a -> Text
 lowShow a = toLower (from (show a))
+
+-- | Note that the size constants for the recursive constructors
+-- are somewhat arbitrary; perhaps we should take into account
```

Comment:
Why arbitrary constants?  Why not just count 1 for each constructor?

Reply:
Sure, that's consistent at least.